lte/gateway/c/oai/tasks/nas5g/include/RegistrationRequest.h

```diff
@@ -0,0 +1,139 @@
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+#pragma once
+#include <sstream>
+#include "ExtendedProtocolDiscriminator.h"
+#include "SecurityHeaderType.h"
+#include "MessageType.h"
+#include "5GSRegistrationType.h"
+#include "NASKeySetIdentifier.h"
+#include "5GSMobileIdentity.h"
+#if 0 // TBD
+#include "5GMMCapability.h"
+#include "UESecurityCapability.h"
+#include "NSSAI.h"
+#include "5GSTrackingAreaIdentity.h"
+#include "S1UENetworkCapability.h"
+#include "UplinkDataStatus.h"
+#include "PDUSessionStatus.h"
+#include "MICOIndication.h"
+#include "UEStatus.h"
+#include "AllowedPDUSessionStatus.h"
+#include "UESUsageSetting.h"
+#include "5GSDRXParameters.h"
+#include "EPSNASMessageContainer.h"
+#include "LADNIndication.h"
+#include "PayloadContainer.h"
+#include"Payload_Container_type.h"
+#include "NetworkSlicingIndication.h"
+#include "5GSUpdateType.h"
+#include "NASMessageContainer.h"
+#include "EPSbearercontextstatus.h"
+#endif
+
+using namespace std;
+namespace magma5g
+{
+  // RegistrationRequest Message Class
+  class RegistrationRequestMsg
+  {
+#define REGISTRATION_REQUEST_MINIMUM_LENGTH 5
+    public:
+      RegistrationRequestMsg();
+      ~RegistrationRequestMsg();
+      ExtendedProtocolDiscriminatorMsg extendedprotocoldiscriminator; 
+      SecurityHeaderTypeMsg securityheadertype; 
+      MessageTypeMsg messagetype; 
+      M5GSRegistrationTypeMsg m5gsregistrationtype;
+      NASKeySetIdentifierMsg naskeysetidentifier; 
+      M5GSMobileIdentityMsg m5gsmobileidentity; 
+#if 0 //TBD
+      M5GMMCapabilityMsg m5gmmcapability; 
+      UESecurityCapabilityMsg uesecuritycapability; 
+      NSSAIMsg nssai; 
+      M5GSTrackingAreaIdentityMsg m5gstrackingareaidentity; 
+      //S1UENetworkCapabilityMsg s1uenetworkcapability; 
+      UplinkDataStatusMsg uplinkdatastatus; 
+      PDUSessionStatusMsg pdusessionstatus; 
+      MICOIndicationMsg micoindication; 
+      UEStatusMsg uestatus; 
+      AllowedPDUSessionStatusMsg allowedpdusessionstatus; 
+      UESUsageSettingMsg uesusagesetting;
+      M5GSDRXParametersMsg m5gsdrxparameters; 
+      EPSNASMessageContainerMsg epsnasmessagecontainer; 
+      LADNIndicationMsg ladnindication; 
+      //PayloadContainerMsg payloadcontainer; 
+      //PayloadContainerTypeMsg payloadcontainertype; 
+      NetworkSlicingIndicationMsg networkslicingindication;  
+      M5GSUpdateTypeMsg m5gsupdatetype;  
+      NASMessageContainerMsg nasmessagecontainer;  
+      //EPSBearerContextStatusMsg epsbearercontextstatus;
+#endif
+      int DecodeRegistrationRequestMsg(RegistrationRequestMsg *registrationrequest, uint8_t *buffer, uint32_t len);
+      int EncodeRegistrationRequestMsg(RegistrationRequestMsg *registrationrequest, uint8_t *buffer, uint32_t len);
+  };
+}
+/*

```

Comment:
can you edit the content below? though it is in comments, it is expected to be formatted.